model/riscv_insts_zfa.sail

```diff
@@ -0,0 +1,767 @@
+/* FLI.H */
+
+union clause ast = RISCV_FLI_H : (bits(5), regidx)
+
+mapping clause encdec = RISCV_FLI_H(rs1, rd)                if (haveFExt() & haveZfa())
+  <-> 0b111_1010 @ 0b00001 @ rs1 @ 0b000 @ rd @ 0b101_0011  if (haveFExt() & haveZfa())
+
+mapping clause assembly = RISCV_FLI_H(constantidx, rd)
+  <-> "fli.h" ^ spc() ^ freg_name(rd) ^ sep() ^ hex_bits_5(constantidx)
+
+function clause execute (RISCV_FLI_H(constantidx, rd)) = {
+  bits : bits(16) = zeros();
+  match constantidx {
+    0b00000 => { bits = 0xbc00 },  /* -1.0 */
+    0b00001 => { bits = 0x0400 },  /* minimum positive normal */
+    0b00010 => { bits = 0x0100 },  /* 1.0 * 2^-16 */
+    0b00011 => { bits = 0x0200 },  /* 1.0 * 2^-15 */
+    0b00100 => { bits = 0x1c00 },  /* 1.0 * 2^-8  */
+    0b00101 => { bits = 0x2000 },  /* 1.0 * 2^-7  */
+    0b00110 => { bits = 0x2c00 },  /* 1.0 * 2^-4  */
+    0b00111 => { bits = 0x3000 },  /* 1.0 * 2^-3  */
+    0b01000 => { bits = 0x3400 },  /* 0.25 */
+    0b01001 => { bits = 0x3500 },  /* 0.3125 */
+    0b01010 => { bits = 0x3600 },  /* 0.375 */
+    0b01011 => { bits = 0x3700 },  /* 0.4375 */
+    0b01100 => { bits = 0x3800 },  /* 0.5 */
+    0b01101 => { bits = 0x3900 },  /* 0.625 */
+    0b01110 => { bits = 0x3a00 },  /* 0.75 */
+    0b01111 => { bits = 0x3b00 },  /* 0.875 */
+    0b10000 => { bits = 0x3c00 },  /* 1.0 */
+    0b10001 => { bits = 0x3d00 },  /* 1.25 */
+    0b10010 => { bits = 0x3e00 },  /* 1.5 */
+    0b10011 => { bits = 0x3f00 },  /* 1.75 */
+    0b10100 => { bits = 0x4000 },  /* 2.0 */
+    0b10101 => { bits = 0x4100 },  /* 2.5 */
+    0b10110 => { bits = 0x4200 },  /* 3 */
+    0b10111 => { bits = 0x4400 },  /* 4 */
+    0b11000 => { bits = 0x4800 },  /* 8 */
+    0b11001 => { bits = 0x4c00 },  /* 16 */
+    0b11010 => { bits = 0x5800 },  /* 2^7 */
+    0b11011 => { bits = 0x5c00 },  /* 2^8 */
+    0b11100 => { bits = 0x7800 },  /* 2^15 */
+    0b11101 => { bits = 0x7c00 },  /* 2^16 */
+    0b11110 => { bits = 0x7c00 },  /* +inf */
+    0b11111 => { bits = canonical_NaN_H() },
+  };
+  F(rd) = nan_box(bits);
```

Comment:
Does this make sense for Zfa, given that in-X is not supported (I am concerned regarding the readability as part of the specifications)?

Or would you prefer us to add new auto-(un)boxing accessors for F (i.e., F_H, F_S, F_D)?

Reply:
Probably we should have non-X variants, though I wouldn't be opposed to just using the X variants everywhere other than for an X<->F move that doesn't make sense on Zfinx, since there's a good chance the instructions will become defined for Zfinx (and there's no reason they shouldn't be).

Reply:
Added the additional accessors w/ auto-(un)boxing.  We can add _or_X accessors back in, if the spec is changed.